lightningd/options.c

```diff
@@ -724,7 +724,7 @@ void register_opts(struct lightningd *ld)
 {
 	opt_set_alloc(opt_allocfn, tal_reallocfn, tal_freefn);
 
-	opt_register_early_noarg("--help|-h", opt_lightningd_usage, ld,
+	opt_register_noarg("--help|-h", opt_lightningd_usage, ld,
 				 "Print this message.");
```

Comment:
This creates a problem; if you run `lightningd --help` it will create the lightning dir before printing anything.  I think you need to leave it as an early arg, but have it set a flag.  Then check if that flag is set and call opt_lightningd_usage after opt parsing is finshed?